### ISSUE TITLE
[css-grid-3][masonry] Initial value of track listing for masonry grid axis

### ISSUE THREAD

**Issue by fantasai** [w3c/csswg-drafts#10869]
The initial value of `grid-template-columns` is `none`, which ends up creating as many tracks as there are template areas, sizing them per `grid-auto-columns`.

Tab's Alternate Masonry Proposal instead uses an initial value of `repeat(auto-areas, auto)` which repeats `auto` to match the template areas (ignoring `masonry-auto-tracks`), and falls back to `repeat(auto-fill, auto)`... which is technically not defined atm, since auto depends on explicit placement.

What should the initial track listing be for masonry layout?

**Comment by tabatkins** [comment]
Yeah, we lost the definition of how to handle intrinsic track sizes during repetition when we figured out the compromise layout algorithm. I've set down what I think is a reasonable definition (basically just restoring the old behavior from my draft) in <https://drafts.csswg.org/css-grid-3/#masonry-intrinsic-repeat>, along with an explanation of *why* the definition is the way it is.

This has been spot-reviewed by @bfgeek, but not yet by the MS engineers actually doing the experimental implementation.

**Comment by fantasai** [comment]
OK, split out the question of whether/how to define auto-repeats into https://github.com/w3c/csswg-drafts/issues/10915 ; leaving this issue to just the question of what the initial value should be (assuming both are possible).

**Comment by alisonmaher** [comment]
If we define a new default value for masonry, we may need to clarify what [`none`](https://www.w3.org/TR/css-grid-2/#valdef-grid-template-none) means in the context of the shorthands. In the case of `grid-template`, for example, `none` "Sets all three properties to their initial values ([none](https://www.w3.org/TR/css-grid-2/#valdef-grid-template-rows-none))."

Would we want the template rows or columns to be set to the new initial value in the case of `masonry`, or `none`?

Also, should this new default value only apply in the grid direction of a masonry (i.e. to `grid-template-columns` in the case of a column masonry, and not in the row direction (which could remain as defaulting to `none`)?

Edit: I suppose the title implies it is only in the grid axis, which is what I would have expected, so nvm on the last question

Further edit: I guess if we are keeping the default as `none` for both and defining that differently for each, then my initial question is a moot point, as well.

**Comment by fantasai** [comment]
Jen and I did a deep dive into the auto-sizing behavior that we'd get from `repeat(auto-fill, auto)` here:
https://webkit.org/blog/16026/css-masonry-syntax/#extending-masonry-with-repeatauto-fill-auto

Our conclusion was that while it's useful in some cases (like a footer sized to the max-width of the longest menu item label), it doesn't actually do anything particularly useful by default. For typical masonry layouts (cards, photos) you have to put some not-obvious sizing controls on the items themselves, at which point it's better for the author to be putting that info into their track sizing declaration directly: they have access to more power that way, and it's more performant for the browser to not have to analyze all the items. For these reasons, we think it's better to nudge authors into using track sizing rather than leading down the path of trying to influence track sizing by sizing the items, which is what having this kind of automatic default would do.

**Comment by alisonmaher** [comment]
Likely, the usefulness of this value for images comes down to [how we decide to resolve %'s](https://github.com/w3c/csswg-drafts/issues/12432) in this mode of track sizing.

The initial proposal was to resolve %'s against the container, which would mean in the examples using `width: 100%` would produce one column filling out the entire container. I would argue this would give similar results to what you would get with the current default - which should lead an author to produce a different set of track definitions as they would today.

The following demo is an example of how this works in Chromium today, which implements this as a prototype: https://developer.chrome.com/blog/masonry-update

<img width="558" height="342" alt="Image" src="https://github.com/user-attachments/assets/609cc250-9905-46ec-851a-ee5f0b01344b" />

In this example, there is a div surrounding each image. The div has a `min-width` of 100px and the images have `width: 100%`. This produces a reasonable result.

If the items are not % sized, or have some sort of intrinsic size from its content, like text, `repeat(auto-fill, auto)` would produce a better result out of the gate than the current default.

As a result, I think `repeat(auto-fill, auto)` at worst produces the same result as the current definition for `none`, in other cases it produces a better initial result. In either case, there are times when an author will want to provide a different track sizing definition than the default provided, and I think that should be encouraged, but I don't see a reason to avoid providing a better default track sizing definition out of the gate if we can.
